rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -865,8 +865,10 @@ namespace rsx
 
 		registers[NV4097_SET_ZSTENCIL_CLEAR_VALUE] = 0xffffffff;
 
-		for (auto& tex : fragment_textures) tex.init();
-		for (auto& tex : vertex_textures) tex.init();
+		std::for_each(fragment_textures.begin(), fragment_textures.end(), [](auto& ft) { ft.init(); });
+		std::for_each(vertex_textures.begin(), vertex_textures.end(), [](auto& vt) { vt.init(); });
+		std::for_each(
+		    vertex_arrays_info.begin(), vertex_arrays_info.end(), [](auto& vtx) { vtx.reset(); });
```

Comment:
I prefer to use std::for_each loops when there is no control flow instruction (break/continue) and no container modification (erase, insert...).
With C++17 the algorithms on loop (for_each, transform...) get new overloads by adding an execution policy : http://en.cppreference.com/w/cpp/algorithm  which gives an hint to the compiler on the iteration order. Hopefully it can help compiler to vectorize code.